Change SQL Server codenames to version names

Wraith2 opened the pull request:
This changes the sql server codenames found in the code to the released version numbers as found at https://en.wikipedia.org/wiki/List_of_Microsoft_codenames#SQL_Server_family

I thought this would be a fairly small set of changes when i started but it has ended up touching a lot of files. There are no functional changes.

Wraith2 commented:
CI doesn't like this but as far as I can see none of the errors are due to my changes.

Kaur-Parminder commented on src/Microsoft.Data.SqlClient/netfx/src/Microsoft/Data/SqlClient/SqlBuffer.cs:
```diff
@@ -1309,5 +1342,59 @@ private void ThrowIfNull()
+        {
+            ThrowIfNull();
+            return (T)(object)_value._single;
+        }
```
@Wraith2 I think you have these changes in two places 
https://github.com/dotnet/SqlClient/pull/1348/files

Kaur-Parminder replied:
@JRahnama Other PR is approved, if we can merge that it will be easier to resolve conflicts of that PR in here.

Wraith2 replied:
rebased to current master , that should resolve this.

Kaur-Parminder commented on src/Microsoft.Data.SqlClient/netfx/src/Resources/Strings.Designer.cs:
```diff
@@ -10539,6 +10557,15 @@ internal class Strings {
+        internal static string SqlConnection_Asynchronous {
+            get {
+                return ResourceManager.GetString("SqlConnection_Asynchronous", resourceCulture);
+            }
```
These don't have name changes are these intended to be in this PR?

JRahnama commented:
Can you address the conflict here please?

Kaur-Parminder commented on src/Microsoft.Data.SqlClient/netfx/src/Microsoft.Data.SqlClient.csproj:
```diff
@@ -338,6 +338,7 @@
     </Compile>
     <Compile Include="..\..\src\Microsoft\Data\SqlClient\SqlCommandBuilder.cs">
       <Link>Microsoft\Data\SqlClient\SqlCommandBuilder.cs</Link>
+      <SubType>Component</SubType>
     </Compile>
```
```suggestion
      <Link>Microsoft\Data\SqlClient\SqlCommandBuilder.cs</Link>
    </Compile>
```

Wraith2 replied:
Done.

Kaur-Parminder commented on src/Microsoft.Data.SqlClient/netfx/src/Resources/Strings.Designer.cs:
```diff
@@ -5550,6 +5550,15 @@ internal class Strings {
+            get {
+                return ResourceManager.GetString("DbConnectionString_AsynchronousProcessing", resourceCulture);
+            }
+        }
```
I could not see the usage of this newly added string and the other one SQL_AsyncConnectionRequired.

Wraith2 replied:
Removed, and I don't quite understand why it was added. It feels like the base branch for this wasn't main or something like that.

Wraith2 commented:
I changed SqlCommandBuilder to have the `DesignerCategory("")` which prevents the SubType coming back. I also reverted the auto expansion of the resx localized imports though the designer keeps wanting to put them back which is unhelpful.

DavoudEshtehari commented on src/Microsoft.Data.SqlClient/netfx/src/Resources/Strings.Designer.cs:
```diff
@@ -39,7 +39,7 @@ internal class Strings {
             get {
                 if (object.ReferenceEquals(resourceMan, null)) {
-                    global::System.Resources.ResourceManager temp = new global::System.Resources.ResourceManager("Microsoft.Data.SqlClient.Resources.Strings", typeof(Strings).Assembly);
+                    global::System.Resources.ResourceManager temp = new global::System.Resources.ResourceManager("SqlClient.Resources.Strings", typeof(Strings).Assembly);
```
Failures should came from these change!
```suggestion
                    global::System.Resources.ResourceManager temp = new global::System.Resources.ResourceManager("Microsoft.Data.SqlClient.Resources.Strings", typeof(Strings).Assembly);
```

Wraith2 replied:
Fixed. The resource naming needs to be looked at sometime in a far future when we have time for such small issues.

DavoudEshtehari replied:
Seems it hasn't been fixed! 😉

Wraith2 replied:
Nope. And I don't understand why it's broken or how i broke it in this PR. I may have to close and start again on a clean branch.

Wraith2 commented:
I don't understand the source of problems in this PR so I'm going to close and rework clearly on a new branch. Was there any comment or issue on the renames themselves?

DavoudEshtehari commented:
As my understanding if you follow my code suggestion it'll be fixed. Please, give it a try before making a new PR.

Wraith2 commented:
I tried, it didn't. That combined with the earlier changes that were resolved by rebasing concerned me enough that I'd rather do it again cleanly than risk an unclean branch causing later issues.
